frontend/src/metabase/css/dashboard.css

```diff
@@ -101,6 +101,14 @@
   z-index: 2;
 }
 
+/* dashcards use positioned absolutely,
+   so children calculate z-index inside a container, not globally,
+   that's why we have to override parent's z-index
+*/
+.DashCard:hover {
+  z-index: 3;
+}
```

Comment:
Clever, I like the idea. This is solution is really simple and I like it. However, let's not add anything to CSS files as they're deprecated.

I think the class name is added here
https://github.com/metabase/metabase/blob/8e1f884aa2d1a785efb22a46e1dca8cb0e3fb56c/frontend/src/metabase/dashboard/components/DashboardGrid.jsx#L401-L407

`<DashboardCard />` is already an emotion component, you should be able to put this CSS rule there instead.

As for other works regarding CSS files, I've seen.

Splitting CSS files or removing rules are fine, but adding one will be questionable.

Reply:
I kinda agree with you that we shouldn't use css files at all, at the same time without the one large refactoring of dashcards css I doubt splitting it between css file and styled file will add any benefit. Let me check how complicated removal of css file will be

Reply:
Fair, if it's not straightforward. I think we can justify leaving it here in this case.

Reply:
I moves some styles, which was easy to test